src/apollo/client.ts

```diff
@@ -74,3 +69,42 @@ export const carrotSubgraphClient: { [chainId: number]: ApolloClient<NormalizedC
     cache: new InMemoryCache(),
   }),
 }
+
+export const subgraphBlocksClientsUris: { [chainId in SWPRSupportedChains]: string } = {
+  [ChainId.MAINNET]: 'https://api.thegraph.com/subgraphs/name/blocklytics/ethereum-blocks',
+  [ChainId.ARBITRUM_ONE]: 'https://api.thegraph.com/subgraphs/name/dodoex/arbitrum-one-blocks',
+  [ChainId.GNOSIS]: 'https://api.thegraph.com/subgraphs/name/1hive/xdai-blocks',
+  // testnests
+  [ChainId.RINKEBY]: 'https://api.thegraph.com/subgraphs/name/blocklytics/ethereum-blocks',
+  [ChainId.ARBITRUM_RINKEBY]: 'https://api.thegraph.com/subgraphs/name/dodoex/arbitrum-one-blocks',
+  [ChainId.ARBITRUM_GOERLI]: '', // FIXME: fix this once the subgraph is deployed
+}
+
+export const subgraphBlocksClients: {
+  [chainId in SWPRSupportedChains]: ApolloClient<NormalizedCacheObject>
+} = {
+  [ChainId.MAINNET]: new ApolloClient({
+    uri: subgraphBlocksClientsUris[ChainId.MAINNET],
+    cache: new InMemoryCache(),
```

Comment:
optional: don't you want to make this pattern into a function and apply it to a list of ChainIds?

Reply:
hmm.. I'll take a look 👍

Reply:
Updated it. Thanks for the tip, much better now.